docs/enhancements/clustersync_data_breakout.md

```diff
@@ -0,0 +1,50 @@
+# Move SyncSet Status out of Etcd
+
+
+## Summary
+The purpose of this enhancement is to break out SyncSet status out of Etcd and move to external storage
+
+## Motivation
+Once a cluster is installed, ClusterSync account for a bulk of Etcd's traffic and storage since it stores per cluster
+status for all matching SyncSets. It can easily be reconstructed if lost, and it is not a resource we reconcile on.
+In order to avoid such regular large writes to Etcd, Hive SyncSet status is the perfect candidate to move out of Etcd.
+This can help Hive clusters scale to the anticipated 1000 per Hive target.
+
+### Goals
+Deploy an alternative storage backend via the Hive Operator and eliminate the ClusterSync Etcd storage.
+
+### Non-Goals
+None
+
+## Open Questions
+1. How to get the details of failing SyncSets for a cluster? Some options to consider:
+    1. Add a small REST API and possibly CLI tool to scrape. However, how to secure the REST API?
```

Comment:
I believe SRE usually gets cluster kubeconfig when there's a problem. How would this theoretical REST API be securely exposed to SRE and not the outside world? Routes would probably be a bad idea. We could do just a Service and require SRE to use kubectl port-forward? Is this secure enough?

How would SRE obtain a CLI tool when they need it? There is some precedent for them checking out hive and running a script. Maybe building a go binary?

Would they be better off with curl/httpie and forget the CLI?

What could the REST API endpoints look like?

What input parameters are needed and how would SRE obtain them?

Reply:
HIve clusters are not on public internet, is routing actually ok? SRE would then sshuttle and curl something like https://hive-aux-api.hivep01s01.openshift.com/clustersync/ ?